frontend/packages/metal3-plugin/src/components/modals/StartNodeMaintenanceModal.tsx

```diff
@@ -20,11 +11,25 @@ import {
 import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
 import { CephClusterModel } from '@console/ceph-storage-plugin/src/models';
 import { startNodeMaintenance } from '../../k8s/requests/node-maintenance';
-import { CEPH_FLAG } from '../../features';
-import { RootState } from '@console/internal/redux';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+const cephClusterResource = {
+  kind: referenceForModel(CephClusterModel),
+  namespaced: false,
+  isList: true,
+};
+
+export type StartNodeMaintenanceModalProps = {
```

Comment:
you could reuse `HandlePromiseProps` and `ModalComponentProps`

```suggestion
export type StartNodeMaintenanceModalProps = HandlePromiseProps & ModalComponentProps & {
  nodeName: string;
```